bn-apps/cordapp-updates-distribution/corda-updates-shell/src/main/kotlin/net/corda/businessnetworks/cordaupdates/shell/ConsoleTransferListener.kt

```diff
@@ -0,0 +1,123 @@
+package net.corda.businessnetworks.cordaupdates.shell
+
+import org.eclipse.aether.transfer.AbstractTransferListener
+import org.eclipse.aether.transfer.MetadataNotFoundException
+import org.eclipse.aether.transfer.TransferEvent
+import org.eclipse.aether.transfer.TransferResource
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Simple maven resolver transfer listener that logs everything into the console
+ * The implementation is kindly borrowed from https://github.com/eclipse/aether-demo/blob/master/aether-demo-snippets/src/main/java/org/eclipse/aether/examples/util/ConsoleTransferListener.java
+ */
+class ConsoleTransferListener : AbstractTransferListener() {
+    companion object {
+        val logger : Logger by lazy { LoggerFactory.getLogger("corda-updates") }
```

Comment:
No need for lazy here. The companion object will itself be lazily initialised on demand.

Reply:
When I remove the lazy bit then the logger gets initialised with a wrong configuration:
Please have a look at how the logger is initialised: https://github.com/corda/corda/blob/master/tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaCliWrapper.kt#L139

Here is how the logger is created inside the `CordaCLIWrapper` itself: https://github.com/corda/corda/blob/master/tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaCliWrapper.kt#L166